tests/h/services/search_index/_queue_test.py

```diff
@@ -348,6 +348,42 @@ def add_all(ids=annotation_ids, schedule_in=MINUS_5_MIN_IN_SECS, force=False):
         return add_all
 
 
+class TestCount:
+    def test_it(self, db_session, factories, now, queue):
+        one_minute = datetime_.timedelta(minutes=1)
+        # 2 jobs that should be counted.
+        factories.Job.create(
+            name="sync_annotation",
+            tag="storage.create_annotation",
+            scheduled_at=now - one_minute,
+        )
+        factories.Job.create(
+            name="sync_annotation",
+            tag="storage.update_annotation",
+            scheduled_at=now - one_minute,
+        )
+        # A job that shouldn't be counted because it has the wrong tag.
+        factories.Job.create(name="sync_annotation", tag="something_else")
+        # A job that shouldn't be counted because it has the wrong name.
+        factories.Job.create(name="something_else", tag="storage.create_annotation")
+        # A job that shouldn't be counted because it's expired.
+        factories.Job.create(
+            name="sync_annotation",
+            tag="storage.create_annotation",
+            expires_at=now - one_minute,
+        )
+        # A job that shouldn't be counted because it isn't scheduled yet.
+        factories.Job.create(
+            name="sync_annotation",
+            tag="storage.create_annotation",
+            scheduled_at=now + one_minute,
+        )
```

Comment:
It's like you want a partial here that creates a good job but accepts arguments:

```
@pytest.fixture
def make_job(
    def make_job(**kwargs):
        return factories.Job.create(
            name=kwargs.get("name", "sync_annotation"), 
            tag=kwargs.get("tag", "storage.update_annotation"),
            schedule_at=kwargs.get("scheduled_at", now - one_minute)
            expires_at=kwargs.get("expires_at", now + one_minute)
        )
    return make_job
```

Then the fixtures here would be something like:

```
# 2 jobs that should be counted.
make_job(tag="matching_tag_1")
make_job(tag="matching_tag_2")

# Jobs that shouldn't be counted
make_job(tag="wrong_tag")
make_job(name="wrong_name")
make_job(expires_at=now - one_minute)
make_job(scheduled_at=now + one_minute)
```

Reply:
factory_boy actually has a way of doing this: you just subclass any factory and add or override any defaults that you want to:

```python
class JobFactory(factories.Job):
    name = "sync_annotation"
    tag = "storage.create_annotation"
    scheduled_at = now - one_minute
```

This works pretty nicely because, like with factories generally, you can pass any arg or kwargs to `JobFactory`, not just ones the factory knows about, and it'll be passed to the model object. It's basically the same as:

```python
def job_factory(**kwargs):
    kwargs.set_default("name", "sync_annotation")
    kwargs.set_default("tag", "storage.create_annotation")
    kwargs.set_default("scheduled_at", now - one_minute)
    return factories.Job(**kwargs)
```

... except that with the factory class version you get all the other features of factory_boy too, should you need them.

I've gone with that, but I've just made it an inline class since it's actually specific to this one particular test.